core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java

```diff
@@ -203,15 +207,23 @@ private void failLateSubscriber(Subscriber<? super T> subscriber, Executor execu
         }
     }
 
-    private void subscribeToDelegate(
-            StreamMessage<T> delegate, PendingSubscription<? super T> pendingSubscription) {
+    private void subscribeToDelegate(StreamMessage<T> delegate,
+                                     PendingSubscription<? super T> pendingSubscription) {
+
+        if (pendingSubscription == NO_OP_PENDING_SUBSCRIPTION ||
```

Comment:
Should these two lines be comparing against `this.pendingSubscription`, not `pendingSubscription`? (renaming the parameter to `newPendingSubscription` would probably make it clearer either way).

Reply:
not two lines. only first statement ?

```
 if (pendingSubscription == NO_OP_ALREADY_SUBSCRIBED ||
            !pendingSubscriptionUpdater.compareAndSet(this,
                                                      newPendingSubscription,
                                                      NO_OP_ALREADY_SUBSCRIBED)) {
            return;
        }
```

because compareAndSet with this.pendingSubscription will always be true, right?